job/job.go

```diff
@@ -3,52 +3,65 @@ package job
 import (
 	"context"
 	"net"
-	"os"
 	"time"
 
 	"github.com/equinix-labs/otel-init-go/otelhelpers"
 	"github.com/packethost/pkg/log"
 	"github.com/pkg/errors"
+	"github.com/tinkerbell/boots/client"
 	"github.com/tinkerbell/boots/conf"
 	"github.com/tinkerbell/boots/dhcp"
-	"github.com/tinkerbell/boots/packet"
-	tw "github.com/tinkerbell/tink/protos/workflow"
 	"go.opentelemetry.io/otel/trace"
 )
 
-var joblog log.Logger
-var client packet.Client
-var provisionerEngineName string
+// JobManager creates jobs.
+type Manager interface {
+	CreateFromRemoteAddr(ctx context.Context, ip string) (context.Context, *Job, error)
+	CreateFromDHCP(context.Context, net.HardwareAddr, net.IP, string) (context.Context, *Job, error)
+}
 
-func Init(l log.Logger) {
-	joblog = l.Package("job")
-	initRSA()
+// Creator is a type that can create jobs
+type Creator struct {
+	finder                client.Finder
+	reporter              client.Reporter
+	provisionerEngineName string
+	logger                log.Logger
+	nowFunc               func() time.Time
 }
 
-// SetClient sets the client used to interact with the api.
-func SetClient(c packet.Client) {
-	client = c
+// NewCreator returns a manager that can create jobs
+func NewCreator(logger log.Logger, provisionerEngineName string, reporter client.Reporter, finder client.Finder) *Creator {
+	return &Creator{
+		finder:                finder,
+		provisionerEngineName: provisionerEngineName,
+		reporter:              reporter,
+		logger:                logger,
+		nowFunc:               time.Now,
```

Comment:
Why do we want a nowFunc here? Its only ever set to `time.Now` so why the indirection?

Reply:
For testability, but a quick search shows that nothing references this, I'll change it back